sparse/impl/KokkosSparse_sptrsv_solve_impl.hpp

```diff
@@ -2609,366 +2450,301 @@ struct TriLvlSchedTP1SingleBlockFunctorDiagValues {
+    auto nodes_grouped_by_level = thandle.get_nodes_grouped_by_level();
+
+    size_type node_count = 0;
+
+    int team_size = thandle.get_team_size();
+    team_size     = team_size == -1 ? 64 : team_size;
+
+    // Start capturing stream
+    if (thandle.cudagraphCreated == false) {
+      Kokkos::fence();
+      cudaStreamBeginCapture(stream1, cudaStreamCaptureModeGlobal);
+      {
+        for (int iter = 0; iter < nlevels; ++iter) {
+          size_type lvl_nodes = hnodes_per_level(iter);
 
-  static inline PolicyType get_policy(int nt, int ts) {
-    return PolicyType(nt, ts);
-  }
+          auto policy = std::is_same<execution_space, Kokkos::Cuda>::value
+                            ? team_policy(lvl_nodes, team_size, cuda1)
+                            : team_policy(lvl_nodes, team_size);
 
-  template <class ExecInstanceType>
-  static inline PolicyType get_policy(int nt, int ts, ExecInstanceType) {
-    return PolicyType(nt, ts);
-    // return PolicyType(ExecInstanceType(),nt,ts);
-  }
-};
-#endif
-#ifdef KOKKOS_ENABLE_OPENMP
-template <>
-struct ReturnRangePolicyType<Kokkos::OpenMP> {
-  using PolicyType = Kokkos::RangePolicy<Kokkos::OpenMP>;
+          Kokkos::parallel_for(
+              "parfor_l_team_cudagraph",
+              Kokkos::Experimental::require(
+                  policy,
+                  Kokkos::Experimental::WorkItemProperty::HintLightWeight),
+              LowerTriLvlSchedTP1SolverFunctor<RowMapType, EntriesType,
+                                               ValuesType, LHSType, RHSType>(
+                  row_map, entries, values, lhs, rhs, nodes_grouped_by_level,
+                  node_count));
 
-  static inline PolicyType get_policy(int nt, int ts) {
-    return PolicyType(nt, ts);
-  }
+          node_count += hnodes_per_level(iter);
+        }
+      }
+      cudaStreamEndCapture(stream1, &graph);
 
-  template <class ExecInstanceType>
-  static inline PolicyType get_policy(int nt, int ts, ExecInstanceType) {
-    return PolicyType(nt, ts);
-    // return PolicyType(ExecInstanceType(),nt,ts);
-  }
-};
-#endif
-#ifdef KOKKOS_ENABLE_CUDA
-template <>
-struct ReturnRangePolicyType<Kokkos::Cuda> {
-  using PolicyType = Kokkos::RangePolicy<Kokkos::Cuda>;
+      // Create graphExec
+      cudaGraphInstantiate(&(lcl_cudagraph->cudagraphinstance), graph, NULL,
+                           NULL, 0);
+      thandle.cudagraphCreated = true;
+    }
+    // Run graph
+    Kokkos::fence();
+    cudaGraphLaunch(lcl_cudagraph->cudagraphinstance, stream1);
 
-  static inline PolicyType get_policy(int nt, int ts) {
-    return PolicyType(nt, ts);
-  }
+    cudaStreamSynchronize(stream1);
+    Kokkos::fence();
+  }  // end lower_tri_solve_cg
 
-  template <class ExecInstanceType>
-  static inline PolicyType get_policy(int nt, int ts, ExecInstanceType stream) {
-    return PolicyType(stream, nt, ts);
-  }
-};
-#endif
-#ifdef KOKKOS_ENABLE_HIP
-template <>
-struct ReturnRangePolicyType<Kokkos::HIP> {
```

Comment:
@lucbv , one thing I found a bit odd: ReturnRangePolicyType (which is unused) provided a specialization for HIP that passes the stream to the policy constructor. ReturnTeamPolicyType does not do this, only providing a specialization for CUDA.